src/pybind/mgr/mgr_module.py

```diff
@@ -512,6 +512,28 @@ def check(*args: Any, **kwargs: Any) -> Tuple[int, str, str]:
         return check
     return CheckFileInput
 
+# If the mgr loses its lock on the database because e.g. the pgs were
+# transiently down, then close it and allow it to be reopened.
+MAX_DBCLEANUP_RETRIES = 3
+def MgrModuleRecoverDB(func: abc.Callable) -> abc.Callable:
+    @functools.wraps(func)
+    def check(self: MgrModule, *args: Any, **kwargs: Any) -> Any:
+        retries = 0
+        while True:
+            try:
+                return func(self, *args, **kwargs)
+            except sqlite3.DatabaseError as e:
```

Comment:
Is `sqlite3.DatabaseError` raised when the client's DB client gets blocklisted?

Reply:
Yes. It shows up as an EIO error.